nexus/src/db/datastore/volume.rs

```diff
@@ -447,6 +467,195 @@ impl DataStore {
                 }
             })
     }
+
+    // Here we remove the read only parent from volume_id, and attach it
+    // to temp_volume_id.
+    //
+    // As this is part of a saga, it will be able to handle being replayed
+    // If we call this twice, any work done the first time through should
+    // not happen again, or be undone.
+    pub async fn volume_remove_rop(
+        &self,
+        volume_id: Uuid,
```

Comment:
Ideally, this function would accept an `opctx: &OpContext` argument and you'd do an authz check in here.  I'm not sure it's worth doing this right now.  You'd have to:

- take the OpContext from the internal API HTTP request and store the serialized version into the saga parameters (see https://github.com/oxidecomputer/omicron/blob/fe04dbec75d101adb655957da677ea920e104d6b/nexus/src/app/instance.rs#L124 for an example)
- in the saga action, make a new OpContext using that serialized one (see https://github.com/oxidecomputer/omicron/blob/fe04dbec75d101adb655957da677ea920e104d6b/nexus/src/app/sagas/instance_create.rs#L419-L420 for an example)
- pass that into this function
- change pool -> pool_authorized
- do some other authz check here
    - You could do something generic like `opctx.authorize(authz::Action::Modify, authz::FLEET)`.  It's not very specific but better than nothing.
    - Otherwise you'd want to define a new "Volume" authz resource.  I don't think we have any other resources for the internal API.  The external API ones live in nexus/src/authz/api_resources.rs and are declared using the `authz_resource!` macro.

I'm not sure if this is worth your while right now or not.  The benefit is that it would be hard for someone to accidentally use this function from code path that really shouldn't be able to do this (e.g., an _external_ API request).

Reply:
Going to thank the review for their good suggestion, then consider implementing it at some later time.